core/frontend/src/components/ethernet/InterfaceCard.vue

```diff
@@ -127,8 +135,58 @@ export default Vue.extend({
     is_static_ip_present(): boolean {
       return this.adapter.addresses.some((address) => address.mode === AddressMode.unmanaged)
     },
+    is_interface_last_ip_address(): boolean {
+      return this.adapter.addresses.length === 1
+    },
+  },
+  mounted() {
+    beacon.registerBeaconListener(this)
   },
   methods: {
+    /**
+     * Opens a dialog and requests the user to confirm the deletion of the IP address.
+     * @returns {Promise<boolean>} - Resolves to true if no confirmation is needed or
+     * granted and false otherwise.
+     */
+    async confirm_last_interface_ip(): Promise<boolean> {
+      if (this.is_interface_last_ip_address) {
+        const dialog = this.$refs.deletionDialog as InstanceType<typeof AddressDeletionDialog>
+
+        dialog.title = 'Last IP Address'
+        dialog.message = 'This is the last IP address on the interface. Are you sure you want to proceed?'
+
+        this.show_deletion_dialog = true
+
+        return new Promise((resolve) => {
+          dialog.resolveCallback = resolve
+        })
```

Comment:
Can we encapsulate this under a single function to avoid the code duplication ?
Btw, since we are using dialog reference, isn't possible for it to provide a function that will abstract all the necessary configurations (title, message, resolveCallback) ?